core/src/main/scala/cats/collections/BitSet.scala

```diff
@@ -739,6 +889,50 @@ object BitSet {
           rhs & this
       }
 
+    def ^(rhs: BitSet): BitSet =
+      rhs match {
+        case Leaf(o, values2) =>
+          if (this eq rhs) {
+            newEmpty(offset)
```

Comment:
(This would imply that we have some bugs in `&` since we return `Empty` in a recursive context there. I'm going to see if I can construct a situation that demonstrates the bug. If so, I'll fix it. If not, I'll reevaluate these comments.)

Reply:
So just to follow up on this:

We don't currently have bugs because our top-level bitsets always are anchored at `offset=0`. (See `BitSet.apply` for details.) You could imagine optimizing these constructors to shift offset forward based on what values we're actually trying to store. If we did that, then I think we'd need to be really uptight about using `newEmpty(offset)` instead of `Empty`.

Right now using `Empty` lets the laws pass (and is slightly more efficient).